src/util/bdd.h

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************\
+
+Module: Binary Decision Diagrams
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Interface to Cudd BDD functions that are used in CBMC
+/// To ensure we can easily switch from Cudd to other BDD libraries if needed,
+/// the BDD functions should only be accessed from this header file.
+
+#ifndef CPROVER_UTIL_BDD_H
+#define CPROVER_UTIL_BDD_H
```

Comment:
❓ 💡 🚫 This should rather be put this into `solvers/bdd`. However, this would require `guardt` (currently in `util`) to be moved somewhere else. `guardt` is used by `goto-symex` and `pointer-analysis`. 
I see the following options: Move `guardt` into
- `solvers/guard`
- `pointer-analysis`
- new module?

Reply:
I'm not sure BDDs should go to `solvers` because they can do more than solving system of equations. They can be use as an alternative to exprt for representing Boolean expressions, so I find `util` appropriate.

Reply:
If you place it into `util` you cannot provide an implementation using miniBDD without moving miniBDD into `util` or creating a new module containing BDD things on which `util` depends .